nbbuild/antsrc/org/netbeans/nbbuild/CustomJavac.java

```diff
@@ -252,4 +276,144 @@ private static boolean isIgnored(
+                }
+                URLClassLoader loader = new NbJavacLoader(urls.toArray(new URL[0]), CustomJavac.class.getClassLoader().getParent());
+                final Class<?> newCompilerClass = Class.forName(MAIN_COMPILER_CLASS, true, loader);
+                assertIsolatedClassLoader(newCompilerClass, loader);
+                System.getProperties().put(MAIN_COMPILER_CLASS, newCompilerClass);
+                c = newCompilerClass;
+            }
+            return (Class<?>) c;
+        }
+
+        private static void assertIsolatedClassLoader(Class<?> c, URLClassLoader loader) throws ClassNotFoundException, BuildException {
+            if (c.getClassLoader() != loader) {
+                throw new BuildException("Class " + c + " loaded by " + c.getClassLoader() + " and not " + loader);
+            }
+            Class<?> stdLoc = c.getClassLoader().loadClass(StandardLocation.class.getName());
+            if (stdLoc.getClassLoader() != loader) {
+                throw new BuildException("Class " + stdLoc + " loaded by " + stdLoc.getClassLoader() + " and not " + loader);
+            }
+        }
+
+        @Override
+        protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+            if (isNbJavacClass(name)) {
+                Class<?> clazz = priorityLoaded.get(name);
+                if (clazz == null) {
+                    clazz = findClass(name);
+                    priorityLoaded.put(name, clazz);
+                }
+                return clazz;
+            }
+            return super.loadClass(name, resolve);
+        }
+
+        @Override
+        public Class<?> loadClass(String name) throws ClassNotFoundException {
+            return loadClass(name, false);
+        }
+
+        private static boolean isNbJavacClass(String name) {
+            return name.startsWith("javax.annotation.")
+                    || name.startsWith("javax.tools.")
+                    || name.startsWith("javax.lang.model.")
+                    || name.startsWith("com.sun.source.")
+                    || name.startsWith("com.sun.tools.");
+        }
+    }
+
+    private static final class NbJavacCompiler extends Javac13 {
+
+        private final Class<?> mainClazz;
+
+        NbJavacCompiler(Class<?> mainClazz) {
+            this.mainClazz = mainClazz;
+        }
+
+        @Override
+        public boolean execute() throws BuildException {
+            attributes.log("Using modern compiler", Project.MSG_VERBOSE);
+            Commandline cmd = setupModernJavacCommand();
+            final String[] args = cmd.getArguments();
+            boolean bootClasspath = false;
+            for (int i = 0; i < args.length; i++) {
+                if (args[i].startsWith("-Xbootclasspath/p:")) { // ide/html
+                    bootClasspath = true;
+                }
+                if (args[i].startsWith("-J")) {
+                    args[i] = "-Xlint:none"; // webcommon/javascript2.editor
```

Comment:
This looks strange. I see the "javac.compilerargs=-J-Xmx512m" and `build.compiler=extJavac` in `project.properties` from `webcommon/javascript2.editor`. Both additions look doubious and most probably should be dropped instead of adding this hack (they are identically present in `php/php.editor`).

Reply:
Thank you. There were two definitions of `javac.compilerargs` in each of the project files. Removed together with the `CustomJavac` hack to eliminate `-J` options in 0112550f0a792